packages/@aws-cdk-testing/framework-integ/test/aws-opensearchservice/test/integ.opensearch.custom-kms-key.ts

```diff
@@ -11,7 +11,26 @@ class TestStack extends Stack {
   constructor(scope: Construct, id: string, props?: StackProps) {
     super(scope, id, props);
 
-    const key = new kms.Key(this, 'Key');
+    // Create a custom policy document without the account root principal
+    const customPolicy = new iam.PolicyDocument({
+      statements: [
+        new iam.PolicyStatement({
+          actions: ['kms:*'],
+          resources: ['*'],
+          principals: [
+            // Add only the specific principals you want to have access
+            // For example, a specific IAM role:
+            new iam.ArnPrincipal('arn:aws:iam::123456789012:role/YourSpecificRole'),
+            // Or a service principal:
+            new iam.ServicePrincipal('lambda.amazonaws.com'),
+          ],
+        }),
+      ],
+    });
+
+    const key = new kms.Key(this, 'Key', {
+      policy: customPolicy,
+    });
```

Comment:
Is this relevant to the TLS change?

Reply:
No, but our security guardian is failing due to account root principal. Without this, our security guardian will fail the check.

Reply:
Can we put this into a separate PR, then? Fixes like these should be clean (i.e. no additional changes) to make reverting without impacting other changes easier.

Reply:
@vishaalmehrishi 

I am fine to revert this change but security guardian will be failing. Will the failing security guardian check block the PR approval?